soak/Dockerfile

```diff
@@ -0,0 +1,29 @@
+FROM python:3.8-alpine
+
+# Persist build arguments into environment variables
+ARG AWS_SERVICE
+ARG E2E_GIT_REF=main
+ARG CONTROLLER_E2E_PATH=./${AWS_SERVICE}-controller/test/e2e
+ENV AWS_SERVICE ${AWS_SERVICE}
+ENV PYTHONPATH ${CONTROLLER_E2E_PATH}
+ENV CONTROLLER_E2E_PATH ${CONTROLLER_E2E_PATH}
+
+WORKDIR /
+# Install dependencies for soak test environment
+RUN apk add --no-cache git bash
+
+# Install python dependencies for running ack e2e tests
+COPY requirements.txt .
+RUN pip install -r requirements.txt
+
+# Copy the script to run soak tests.
+COPY soak/run_soak_test.sh .
+RUN chmod +x run_soak_test.sh
+
+# Checkout the controller repository where e2e tests are present.
+# Soak test run consists of multiple runs of these e2e tests.
+RUN git clone https://github.com/aws-controllers-k8s/${AWS_SERVICE}-controller.git
+RUN cd ${AWS_SERVICE}-controller && git checkout ${E2E_GIT_REF}
```

Comment:
Git clone seems to offer the ability to choose the branch ? https://git-scm.com/docs/git-clone#Documentation/git-clone.txt---branchltnamegt

Can we also do a clone depth of 0?